lib/rules/require-default-props.js

```diff
@@ -17,6 +17,9 @@ const report = require('../util/report');
 const messages = {
   noDefaultWithRequired: 'propType "{{name}}" is required and should not have a defaultProps declaration.',
   shouldHaveDefault: 'propType "{{name}}" is not required, but has no corresponding defaultProps declaration.',
+  noDefaultPropsWithFunction: 'Don\'t use defaultProps with function component.',
+  shouldInitializeOptional: '`propType {{name}} is not required, but doesn\'t initialize.`',
```

Comment:
```suggestion
  shouldInitializeOptional: '`propType {{name}} is not required, but doesn’t initialize.`',
```
also not confident about the wording here

Reply:
What's mean that not confident about the wording?

Reply:
Like I’m not sure if this is the best way to word the message, but i don’t have a concrete suggestion yet.

Reply:
I refactor that and option name. How about this?